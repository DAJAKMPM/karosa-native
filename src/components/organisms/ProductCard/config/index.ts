/**
 *
 * @format
 *
 */

export const RATING_COUNT = 5;
export const IMAGE_SIZE = 15;
export const NUM_LINES = 2;
