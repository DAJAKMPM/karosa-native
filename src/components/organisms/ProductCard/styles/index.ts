/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme, DIMENS, SPACING } from "@app/styles";

const ProductCardStyles = StyleSheet.create({
  mainContainer: {
    height: 285,
    width: DIMENS.screenWidth * 0.45,
    margin: 0,
    padding: 0,
    shadowOpacity: 0.03,
    shadowOffset: { height: 2, width: 1 },
    shadowColor: theme.colors.black,
    borderRadius: 3,
    position: "relative",
    zIndex: 1,
  },
  variationThreeCard: {
    height: 200,
    width: DIMENS.screenWidth * 0.4,
    margin: 0,
    padding: 0,
    shadowOpacity: 0.08,
    shadowOffset: { height: 3.4, width: 0 },
    shadowColor: theme.colors.black,
    borderRadius: 3,
    position: "relative",
    zIndex: 1,
  },
  wholesaleContainer: {
    backgroundColor: theme.colors.primary,
    position: "absolute",
    padding: 2,
    zIndex: 99,
    top: 3,
    left: -4,
  },
  ribbonContainer: {
    position: "absolute",
    top: 0,
    right: 0,
    zIndex: 1,
  },
  bottomContentContainer: {
    paddingTop: SPACING.md,
    paddingLeft: SPACING.sm,
    paddingRight: SPACING.sm,
  },
  rowContainer: {
    flexDirection: "row",
    marginVertical: SPACING.sm,
  },
  lastRowContainer: {
    flexDirection: "row",
    marginVertical: SPACING.xs,
  },
  buttonContainer: {
    position: "absolute",
    bottom: -43,
    alignSelf: "center",
  },
  prevPriceContainer: {
    marginLeft: SPACING.xs,
  },
  floatRight: { alignSelf: "center", marginLeft: "auto" },
  buttonContent: {
    backgroundColor: theme.colors.white,
    borderColor: theme.colors.primary,
    minWidth: 124,
    width: 91,
    height: 32,
    padding: 0,
    borderWidth: 2,
  },
  txtButton: {
    ...theme.textLight,
    fontWeight: "700",
    color: theme.colors.primary,
  },
  txtPrice: {
    ...theme.textBold,
    fontWeight: "500",
    fontStyle: "normal",
  },
  txtVariationThreePrice: {
    ...theme.textLight,
    fontWeight: "400",
    fontStyle: "normal",
  },
  txtLocation: {
    ...theme.textLight,
    color: theme.colors.dark30,
    fontWeight: "300",
    fontStyle: "normal",
  },
  txtPrevPrice: {
    ...theme.textRegular,
    color: theme.colors.dark10,
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
    textDecorationColor: theme.colors.dark10,
    fontWeight: "300",
    fontStyle: "normal",
  },
  txtSold: {
    ...theme.textLight,
    fontWeight: "300",
    fontStyle: "normal",
  },
  txtWholesale: {
    ...theme.textLightBold,
    color: theme.colors.white,
    textAlign: "center",
    marginTop: SPACING.xss,
  },
  txtName: {
    ...theme.textRegular,
    fontWeight: "400",
    fontStyle: "normal",
  },
  txtVariationThreeName: {
    ...theme.textLightBold,
    fontWeight: "700",
    fontStyle: "normal",
  },
  image: { width: "100%", height: undefined, aspectRatio: 1.1 },
});

export default ProductCardStyles;
