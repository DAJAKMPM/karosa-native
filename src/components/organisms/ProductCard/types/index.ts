/**
 *
 * @format
 *
 */

export type PropsType = {
  variation: number;
  discount?: string;
  image?: string;
  name: string;
  currentPrice?: string;
  previousPrice?: string;
  sold?: string;
  buttonTitle?: string;
  onButtonClick?(): void;
  rating?: number;
  location?: string;
};
