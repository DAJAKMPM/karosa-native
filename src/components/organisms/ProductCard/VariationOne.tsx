/**
 *
 * VariantOne
 * @format
 *
 */

import React, { FC } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";
import Image from "@app/atoms/Image";
import Card from "@app/molecules/Card";
import Ribbon from "@app/atoms/Ribbon";
import Button from "@app/atoms/Button";

import type { PropsType } from "./types";
import { NUM_LINES } from "./config";
import ProductCardStyles from "./styles";

const VariantOne: FC<PropsType> = (props) => {
  const {
    name,
    image,
    buttonTitle,
    onButtonClick,
    sold,
    currentPrice,
    discount,
  } = props;

  return (
    <Card wrapperStyle={ProductCardStyles.mainContainer}>
      <View style={ProductCardStyles.wholesaleContainer}>
        <Text customStyle={ProductCardStyles.txtWholesale} text={"Wholesale"} />
      </View>
      {discount && (
        <View style={ProductCardStyles.ribbonContainer}>
          <Ribbon upperText={discount} lowerText="OFF" />
        </View>
      )}
      <Image
        source={{ uri: image }}
        customStyle={ProductCardStyles.image}
        resizeMode={"cover"}
      />
      <View style={ProductCardStyles.bottomContentContainer}>
        <Text
          customStyle={ProductCardStyles.txtName}
          numberOfLines={NUM_LINES}
          text={name}
        />
        <View style={ProductCardStyles.rowContainer}>
          <Text
            customStyle={ProductCardStyles.txtPrice}
            text={`P${currentPrice}`}
          />
          <View style={ProductCardStyles.floatRight}>
            <Text
              customStyle={ProductCardStyles.txtSold}
              text={`${sold} sold`}
            />
          </View>
        </View>
      </View>
      <View style={ProductCardStyles.buttonContainer}>
        <Button
          type="outline"
          title={buttonTitle ? buttonTitle : ""}
          customStyle={ProductCardStyles.buttonContent}
          titleStyle={ProductCardStyles.txtButton}
          onPress={onButtonClick ? onButtonClick : () => 0}
        />
      </View>
    </Card>
  );
};

export default VariantOne;
