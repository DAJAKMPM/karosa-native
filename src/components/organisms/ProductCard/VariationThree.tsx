/**
 *
 * VariantTwo
 * @format
 *
 */

import React, { FC, useState } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";
import Image from "@app/atoms/Image";
import Rating from "@app/atoms/Rating";
import Icon from "@app/atoms/Icon";
import Ribbon from "@app/atoms/Ribbon";
import Card from "@app/molecules/Card";

import type { PropsType } from "./types";
import { IMAGE_SIZE, NUM_LINES, RATING_COUNT } from "./config";
import ProductCardStyles from "./styles";

const VariantThree: FC<PropsType> = (props) => {
  const {
    name,
    wholesale,
    image,
    rating,
    location,
    sold,
    currentPrice,
    discount,
    previousPrice,
    onHeartClick,
    heartFlag,
  } = props;

  const [heartIcon, setHeartIcon] = useState(heartFlag);

  const toggleHeartIcon = () => {
    if (heartIcon) {
      setHeartIcon(false);
    } else {
      setHeartIcon(true);
    }
  };

  return (
    <Card containerStyle={ProductCardStyles.mainContainer}>
      {wholesale && (
        <View style={ProductCardStyles.wholesaleContainer}>
          <Text textStyle={ProductCardStyles.txtWholesale} text={"Wholesale"} />
        </View>
      )}

      {discount && (
        <View style={ProductCardStyles.ribbonContainer}>
          <Ribbon upperText={`${discount}%`} lowerText="OFF" />
        </View>
      )}
      <Image
        source={{ uri: image }}
        imageStyle={ProductCardStyles.image}
        resizeMode={"cover"}
      />
      <View style={ProductCardStyles.bottomContentContainer}>
        <Text
          textStyle={ProductCardStyles.txtName}
          numberOfLines={NUM_LINES}
          text={name}
        />
        <View style={ProductCardStyles.rowContainer}>
          <Text
            textStyle={ProductCardStyles.txtPrice}
            text={`P${currentPrice}`}
          />
          <View style={ProductCardStyles.prevPriceContainer}>
            <Text
              textStyle={ProductCardStyles.txtPrevPrice}
              text={`P${previousPrice}`}
            />
          </View>
        </View>
        <View style={ProductCardStyles.rowContainer}>
          <Rating
            ratingCount={RATING_COUNT}
            readonly
            imageSize={IMAGE_SIZE}
            startingValue={rating}
          />
          <View style={ProductCardStyles.floatRight}>
            <Text textStyle={ProductCardStyles.txtSold} text={`${sold} sold`} />
          </View>
        </View>
        <View style={ProductCardStyles.rowContainer}>
          <Text
            textStyle={ProductCardStyles.txtLocation}
            text={location ? location : ""}
          />
          <View
            style={ProductCardStyles.floatRight}
            onTouchStart={toggleHeartIcon}>
            <Icon
              group="common"
              name={heartIcon ? "redHeart" : "outlineHeart"}
              height={12}
              width={13}
            />
          </View>
        </View>
      </View>
    </Card>
  );
};

export default VariantThree;
