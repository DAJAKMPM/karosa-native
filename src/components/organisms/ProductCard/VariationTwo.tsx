/**
 *
 * VariationTwo
 * @format
 *
 */

import React, { FC } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";
import Image from "@app/atoms/Image";
import Rating from "@app/atoms/Rating";
import Icon from "@app/atoms/Icon";
import Ribbon from "@app/atoms/Ribbon";
import Card from "@app/molecules/Card";

import type { PropsType } from "./types";
import { IMAGE_SIZE, NUM_LINES, RATING_COUNT } from "./config";
import ProductCardStyles from "./styles";

const VariationTwo: FC<PropsType> = (props) => {
  const {
    name,
    wholesale,
    image,
    rating,
    location,
    sold,
    currentPrice,
    discount,
    previousPrice,
    onPress,
    ratingImage,
    ratingType,
  } = props;

  return (
    <Card containerStyle={ProductCardStyles.mainContainer} onPress={onPress}>
      {wholesale && (
        <View style={ProductCardStyles.wholesaleContainer}>
          <Text textStyle={ProductCardStyles.txtWholesale} text={"Wholesale"} />
        </View>
      )}

      {discount && (
        <View style={ProductCardStyles.ribbonContainer}>
          <Ribbon upperText={`${discount}%`} lowerText="OFF" />
        </View>
      )}

      <Image
        source={{ uri: image }}
        imageStyle={ProductCardStyles.image}
        resizeMode="cover"
      />

      <View style={ProductCardStyles.bottomContentContainer}>
        <Text
          textStyle={ProductCardStyles.txtName}
          numberOfLines={NUM_LINES}
          text={name}
        />
        <View style={ProductCardStyles.rowContainer}>
          <Text
            textStyle={ProductCardStyles.txtPrice}
            text={`P${currentPrice}`}
          />

          {previousPrice && (
            <View style={ProductCardStyles.prevPriceContainer}>
              <Text
                textStyle={ProductCardStyles.txtPrevPrice}
                text={`P${previousPrice}`}
              />
            </View>
          )}
        </View>
        <View style={ProductCardStyles.rowContainer}>
          <Rating
            ratingCount={RATING_COUNT}
            readonly
            imageSize={IMAGE_SIZE}
            startingValue={rating}
            ratingImage={ratingImage}
            type={ratingType}
          />
          <View style={ProductCardStyles.floatRight}>
            <Text textStyle={ProductCardStyles.txtSold} text={`${sold} sold`} />
          </View>
        </View>
        <View style={ProductCardStyles.lastRowContainer}>
          <Text
            textStyle={ProductCardStyles.txtLocation}
            text={location ? location : ""}
          />
          <View style={ProductCardStyles.floatRight}>
            <Icon group="common" name="outlineHeart" height={12} width={13} />
          </View>
        </View>
      </View>
    </Card>
  );
};

export default VariationTwo;
