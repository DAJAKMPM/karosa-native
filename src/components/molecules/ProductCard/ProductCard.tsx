/**
 *
 * HomeCard
 * @format
 *
 */

import React, { FC } from "react";
import { View, Image } from "react-native";
import { Card } from "react-native-elements";
import { BaseText } from "@app/components/base-text";
import { theme } from "@app/styles";

import type { HomeProps } from "./types";
import ProductCardStyles from "./styles";

const ProductCard: FC<HomeProps> = (props) => {
  const { item } = props;

  return (
    <Card
      containerStyle={{
        height: 220,
        width: 156,
        padding: 0,
        borderRadius: 3,
        shadowOpacity: 0.1,
        shadowOffset: { height: 4, width: 1 },
        shadowColor: theme.colors.black,
      }}>
      <View style={ProductCardStyles.wholesaleContainer}>
        <BaseText style={ProductCardStyles.txtWholesale}>Wholesale</BaseText>
      </View>
      <View style={ProductCardStyles.discount}>
        <View style={ProductCardStyles.discountContainer}>
          <BaseText style={ProductCardStyles.txtDiscount}>30%</BaseText>
          <View style={ProductCardStyles.space} />
          <BaseText style={ProductCardStyles.txtDiscount}>OFF</BaseText>
        </View>
        <View style={ProductCardStyles.triangleLeft} />
        <View style={ProductCardStyles.triangleRight} />
      </View>
      <Image
        source={require("../../../../assets/hinata.jpg")}
        style={ProductCardStyles.image}
        resizeMode={"cover"}
      />
      <View style={ProductCardStyles.mainTxtContainer}>
        <BaseText style={ProductCardStyles.txtName} numberOfLines={2}>
          {item.name}
        </BaseText>
        <View style={ProductCardStyles.txtContainer}>
          <BaseText style={ProductCardStyles.txtPrice}>P300</BaseText>
          <BaseText style={ProductCardStyles.txtSold}>50 sold</BaseText>
        </View>
      </View>
    </Card>
  );
};

export default ProductCard;
