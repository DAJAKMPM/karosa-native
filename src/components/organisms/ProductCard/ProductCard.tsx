/**
 *
 * ProductCard
 * @format
 *
 */

import React, { FC } from "react";

import type { PropsType } from "./types";
import VariationOne from "./VariationOne";
import VariationTwo from "./VariationTwo";

const ProductCard: FC<PropsType> = (props) => {
  const {
    name,
    variation,
    buttonTitle,
    onButtonClick,
    previousPrice,
    image,
    rating,
    location,
    sold,
    currentPrice,
    wholesale,
    discount,
  } = props;

  switch (variation) {
    case 1:
      return (
        <VariationOne
          name={name}
          wholesale={wholesale}
          image={image}
          buttonTitle={buttonTitle}
          onButtonClick={onButtonClick}
          sold={sold}
          currentPrice={currentPrice}
          discount={discount}
          variation={variation}
        />
      );
    case 2:
      return (
        <VariationTwo
          name={name}
          image={image}
          rating={rating}
          location={location}
          sold={sold}
          currentPrice={currentPrice}
          previousPrice={previousPrice}
          discount={discount}
          variation={variation}
        />
      );
    default:
      return (
        <VariationOne
          name={name}
          image={image}
          buttonTitle={buttonTitle}
          onButtonClick={onButtonClick}
          sold={sold}
          currentPrice={currentPrice}
          discount={discount}
          variation={variation}
        />
      );
  }
};

export default ProductCard;
