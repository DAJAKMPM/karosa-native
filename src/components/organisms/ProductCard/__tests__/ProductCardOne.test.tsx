import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import { COMMON } from "@app/constants";
import ProductCard from "../ProductCard";

describe("ProductCard variation one Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <ProductCard name="test" variation={COMMON.VARIATION.ONE} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
