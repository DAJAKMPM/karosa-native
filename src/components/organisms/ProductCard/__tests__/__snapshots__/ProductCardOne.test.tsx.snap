// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard variation one Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginBottom": 0,
        "padding": 0,
        "position": "relative",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.03,
        "shadowRadius": 1,
        "width": 337.5,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "aspectRatio": 1.1,
              "bottom": 0,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1.1,
                "backgroundColor": "#bdbdbd",
                "height": undefined,
                "justifyContent": "center",
                "width": "100%",
              }
            }
            testID="RNE__Image__placeholder"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#1C1C1C"
                hidesWhenStopped={true}
                size="small"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 50,
                "width": 50,
              },
              Object {
                "aspectRatio": 1.1,
                "height": undefined,
                "width": "100%",
              },
            ]
          }
          testID="RNE__Image__children__container"
        />
      </View>
      <View
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 18.75,
            }
          }
        >
          test
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "SFProText-Bold",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "500",
                "lineHeight": 19,
              }
            }
          >
            Pundefined
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginLeft": "auto",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontFamily": "SFProText-Regular",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "300",
                  "lineHeight": 14.06,
                }
              }
            >
              undefined sold
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 8,
            "marginTop": 16,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              undefined,
              Object {
                "backgroundColor": "#fff",
                "overflow": "visible",
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderColor": "#0AA351",
                  "borderRadius": 3,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "height": 32,
                  "justifyContent": "center",
                  "minWidth": 124,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
